src/ggml.c

```diff
@@ -18640,6 +18640,13 @@ struct gguf_context * gguf_init_from_file(const char * fname, struct gguf_init_p
             }
 
             ok = ok && gguf_fread_str(file, &info->name,                          &offset);
+            if (strlen(info->name.data) > GGML_MAX_NAME - 1) {
+                fprintf(stderr, "%s: tensor name '%s' is too long (maximum length: %d characters)\n", __func__, info->name.data, GGML_MAX_NAME - 1);
+                fclose(file);
+                gguf_free(ctx);
+                return NULL;
+            }
+            
```

Comment:
This should be a bit better, since it will reuse the error handling logic below:

```suggestion
            if (strlen(info->name.data) > GGML_MAX_NAME - 1) {
                fprintf(stderr, "%s: tensor name '%s' is too long (maximum length: %d characters)\n", __func__, info->name.data, GGML_MAX_NAME - 1);
                ok = false;
            }
            
            ok = ok && gguf_fread_str(file, &info->name,                          &offset);
```

Reply:
You're totally right!